test/MouseAdapterSpec.js

```diff
@@ -270,5 +270,123 @@ define(function(require) {
                 }]);
             });
         });
+
+        describe('user scroll detection', function() {
+
+            describe('_calculateDeltaDirection', function() {
+                it('should detect an increasing trend in the data', function() {
+                    expect(adapter._calculateDeltaDirection([1, 2, 3, 4, 5])).toBe(1);
+                    expect(adapter._calculateDeltaDirection([1, 1, 1, 2, 2])).toBe(1);
+                    expect(adapter._calculateDeltaDirection([-10, -8, -9, -7, -8])).toBe(1);
+                });
+
+                it('should detect a decreasing trend in the data', function() {
+                    expect(adapter._calculateDeltaDirection([5, 4, 3, 2, 1])).toBe(-1);
+                    expect(adapter._calculateDeltaDirection([2, 2, 1, 1, 1])).toBe(-1);
+                    expect(adapter._calculateDeltaDirection([-8, -7, -9, -8, -10])).toBe(-1);
+                });
+
+                it('should detect a stable trend in the data', function() {
+                    expect(adapter._calculateDeltaDirection([99, 99, 99, 99, 99])).toBe(0);
+                    expect(adapter._calculateDeltaDirection([1, 2, 1, 2, 1])).toBe(0);
+                    expect(adapter._calculateDeltaDirection([10, 9, 5, 9, 10])).toBe(0);
+                });
+
+                it('should begin at the specified offset when passed in', function() {
+                    expect(adapter._calculateDeltaDirection([1, 1, 1, 1, 2, 2, 2, 2], 0)).toBe(1);
+                    expect(adapter._calculateDeltaDirection([1, 1, 1, 1, 2, 2, 2, 2], 4)).toBe(-1);
+                });
+
+                it('should stop at the specified length when passed in', function() {
+                    expect(adapter._calculateDeltaDirection([1, 1, 1, 1, 2, 2, 2, 2], 0, 4)).toBe(0);
+                    expect(adapter._calculateDeltaDirection([1, 1, 1, 1, 2, 2, 2, 2], 4, 4)).toBe(0);
+                });
+
+            });
+
+            describe('_detectNegativeSpike', function() {
+                // Test arrays in this block should be of size MouseAdapter.DELTA_ARRAY_SIZE,
+                // representing mouse event deltas received in chronological order from index 0.
+
+                it('should detect sufficiently steep negative spikes in data', function() {
+                    expect(adapter._detectNegativeSpike([63.5, 62, 59, 110, 49.5, 45, 41, 37.5, 34.5, 0.5])).toBe(true);
+                    expect(adapter._detectNegativeSpike([29, 26.5, 24, 21.5, 20, 18, 32, 14, 13, 0.5])).toBe(true);
+                    expect(adapter._detectNegativeSpike([6, 5.5, 5, 5, 4.5, 4, 3.5, 3, 3, 0.5])).toBe(true);
```

Comment:
I think I see the spikes in the first two (110 and 32, respectively), but where's the spike in this one?

It'd be great to see a comment about how you came up with these exact numbers and how a future dev could come up with her/his own such numbers (without just guessing).